diff -r c489f6ce41c0 GNUmakefile
--- a/GNUmakefile	Mon Oct 22 16:31:52 2012 +0200
+++ b/GNUmakefile	Mon Oct 22 16:32:08 2012 +0200
@@ -18,7 +18,7 @@
 CROSS=
 #CROSS=i686-mingw-
 CC=$(CROSS)gcc
-CFLAGS=-pipe -W -Wall -O2 -fomit-frame-pointer
+CFLAGS=-fPIC -pipe -W -Wall -O2 -fomit-frame-pointer
 #CFLAGS=-pipe -Os -march=pentiumpro -mcpu=pentiumpro -fomit-frame-pointer -fschedule-insns2 -Wall
 
 # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
@@ -27,7 +27,7 @@
 
 # startrip
 ifneq ($(DEBUG),)
-CFLAGS=-pipe -Wall -g
+CFLAGS=-fPIC -pipe -Wall -g
 endif
 path = $(subst :, ,$(PATH))
 diet_path = $(foreach dir,$(path),$(wildcard $(dir)/diet))
@@ -224,8 +224,9 @@
 
 haveepoll.h: tryepoll.c
 	-rm -f $@
-	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c >/dev/null 2>&1; then echo "#define HAVE_EPOLL 1"; else \
-	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c -lepoll >/dev/null 2>&1; then echo "#define HAVE_EPOLL 2"; fi; fi > $@
+	echo > $@
+#	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c >/dev/null 2>&1; then echo "#define HAVE_EPOLL 1"; else \
+#	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c -lepoll >/dev/null 2>&1; then echo "#define HAVE_EPOLL 2"; fi; fi > $@
 	-rm -f tryepoll
 
 havedevpoll.h: trydevpoll.c
@@ -238,7 +239,8 @@
 
 havesigio.h: trysigio.c
 	-rm -f $@
-	if $(DIET) $(CC) $(CFLAGS) -c trysigio.c >/dev/null 2>&1; then echo "#define HAVE_SIGIO"; fi > $@
+	echo > $@
+#	if $(DIET) $(CC) $(CFLAGS) -c trysigio.c >/dev/null 2>&1; then echo "#define HAVE_SIGIO"; fi > $@
 	-rm -f trysigio.o
 
 havealloca.h: tryalloca.c
diff -r c489f6ce41c0 Makefile
--- a/Makefile	Mon Oct 22 16:31:52 2012 +0200
+++ b/Makefile	Mon Oct 22 16:32:08 2012 +0200
@@ -19,7 +19,7 @@
 CROSS=
 #CROSS=i686-mingw-
 CC=$(CROSS)gcc
-CFLAGS=-pipe -W -Wall -O2 -fomit-frame-pointer
+CFLAGS=-fPIC -pipe -W -Wall -O2 -fomit-frame-pointer
 #CFLAGS=-pipe -Os -march=pentiumpro -mcpu=pentiumpro -fomit-frame-pointer -fschedule-insns2 -Wall
 
 # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
@@ -680,8 +680,9 @@
 
 haveepoll.h: tryepoll.c
 	-rm -f $@
-	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c >/dev/null 2>&1; then echo "#define HAVE_EPOLL 1"; else \
-	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c -lepoll >/dev/null 2>&1; then echo "#define HAVE_EPOLL 2"; fi; fi > $@
+	echo > $@
+#	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c >/dev/null 2>&1; then echo "#define HAVE_EPOLL 1"; else \
+#	if $(DIET) $(CC) $(CFLAGS) -o tryepoll tryepoll.c -lepoll >/dev/null 2>&1; then echo "#define HAVE_EPOLL 2"; fi; fi > $@
 	-rm -f tryepoll
 
 havedevpoll.h: trydevpoll.c
@@ -694,7 +695,8 @@
 
 havesigio.h: trysigio.c
 	-rm -f $@
-	if $(DIET) $(CC) $(CFLAGS) -c trysigio.c >/dev/null 2>&1; then echo "#define HAVE_SIGIO"; fi > $@
+	echo > $@
+#	if $(DIET) $(CC) $(CFLAGS) -c trysigio.c >/dev/null 2>&1; then echo "#define HAVE_SIGIO"; fi > $@
 	-rm -f trysigio.o
 
 havealloca.h: tryalloca.c
