--- a/sys/sim/generate-linker-script.py
+++ b/sys/sim/generate-linker-script.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import re
 
--- a/usr.sbin/config/main.c
+++ b/usr.sbin/config/main.c
@@ -121,7 +121,7 @@
 int *
 __error(void)
 {
-  int test = 0;
+  static int test = 0;
   return (&test);
 }
 
--- a/sys/sim/sim-device.c
+++ b/sys/sim/sim-device.c
@@ -525,7 +525,7 @@
 
 static const u_char etherbroadcastaddr[ETHER_ADDR_LEN] =
 			{ 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
-struct SimDevice *sim_dev_create (void *priv, enum SimDevFlags flags)
+struct SimDevice *sim_dev_create (const char *ifname, void *priv, enum SimDevFlags flags)
 {
   /* FIXME!! */
   static uint32_t devnum = 0;
--- a/sys/sim/sim.c
+++ b/sys/sim/sim.c
@@ -79,7 +79,7 @@
 extern void sim_task_destroy (struct SimTask *task);
 extern void *sim_task_get_private (struct SimTask *task);
 
-FORWARDER2(sim_dev_create, nvoid, struct SimDevice *, void *, enum SimDevFlags);
+FORWARDER3(sim_dev_create, nvoid, struct SimDevice *, const char *, void *, enum SimDevFlags);
 FORWARDER1(sim_dev_destroy, void, void, struct SimDevice *);
 FORWARDER2(sim_dev_set_address, void, void, struct SimDevice *, unsigned char *);
 FORWARDER2(sim_dev_set_mtu, void, void, struct SimDevice *, int);
--- a/sys/sim/include/sim-init.h
+++ b/sys/sim/include/sim-init.h
@@ -38,7 +38,7 @@
   void (*sock_poll) (struct SimSocket *socket, void *ret);
   void (*sock_pollfreewait) (void *polltable);
 
-  struct SimDevice *(*dev_create) (void *priv, enum SimDevFlags flags);
+  struct SimDevice *(*dev_create) (const char *ifname, void *priv, enum SimDevFlags flags);
   void (*dev_destroy) (struct SimDevice *dev);
   void *(*dev_get_private) (struct SimDevice *task);
   void (*dev_set_address) (struct SimDevice *dev, unsigned char buffer[6]);
