diff -r 7752dc4ce7e9 src/internet/model/tcp-socket-base.cc
--- a/src/internet/model/tcp-socket-base.cc	Wed Aug 15 18:08:46 2012 -0700
+++ b/src/internet/model/tcp-socket-base.cc	Tue Sep 04 17:50:49 2012 +0200
@@ -1817,11 +1817,12 @@
                     " highestRxAck " << m_txBuffer.HeadSequence () <<
                     " pd->Size " << m_txBuffer.Size () <<
                     " pd->SFS " << m_txBuffer.SizeFromSequence (m_nextTxSequence));
-      // Quit if send disallowed
       if (m_shutdownSend)
         {
-          m_errno = ERROR_SHUTDOWN;
-          return false;
+          // Do not Quit if send disallowed
+          NS_LOG_INFO ("TcpSocketBase::SendPendingData: m_shutdownSend=" << m_shutdownSend);
+        //  m_errno = ERROR_SHUTDOWN;
+        //  return false;
         }
       // Stop sending if we need to wait for a larger Tx window (prevent silly window syndrome)
       if (w < m_segmentSize && m_txBuffer.SizeFromSequence (m_nextTxSequence) > w)
