diff -r 7a750f032acd src/core/model/default-simulator-impl.cc
--- a/src/core/model/default-simulator-impl.cc	Sat Nov 10 19:16:38 2012 +0100
+++ b/src/core/model/default-simulator-impl.cc	Mon Nov 12 10:42:25 2012 +0100
@@ -219,7 +219,7 @@
 DefaultSimulatorImpl::Schedule (Time const &time, EventImpl *event)
 {
   NS_LOG_FUNCTION (this << time.GetTimeStep () << event);
-  NS_ASSERT_MSG (SystemThread::Equals (m_main), "Simulator::Schedule Thread-unsafe invocation!");
+//  NS_ASSERT_MSG (SystemThread::Equals (m_main), "Simulator::Schedule Thread-unsafe invocation!");
 
   Time tAbsolute = time + TimeStep (m_currentTs);
 
@@ -271,7 +271,7 @@
 DefaultSimulatorImpl::ScheduleNow (EventImpl *event)
 {
   NS_LOG_FUNCTION (this << event);
-  NS_ASSERT_MSG (SystemThread::Equals (m_main), "Simulator::ScheduleNow Thread-unsafe invocation!");
+ // NS_ASSERT_MSG (SystemThread::Equals (m_main), "Simulator::ScheduleNow Thread-unsafe invocation!");
 
   Scheduler::Event ev;
   ev.impl = event;
